reflex/utils/format.py

```diff
@@ -449,6 +450,10 @@ def format_state(value: Any) -> Any:
     if isinstance(value, go.Figure):
         return json.loads(to_json(value))["data"]  # type: ignore
 
+    # Convert datetime object to str.
+    if isinstance(value, datetime):
```

Comment:
since `datetime` was added as a `PrimitiveType`, it's getting picked up earlier in this function by `if isinstance(value, types.StateBases)`